### PR TITLE
Update THIRD-PARTY-NOTICES for 8.0 release

### PR DESCRIPTION
**Pull request by NikolaMilosavljevic**
Update of THIRD-PARTY-NOTICES file for 8.0 release.

Generated using infra, command: `build -subset RegenerateThirdPartyNotices`

Full list of repos that contribute to .NET THIRD-PARTY-NOTICES.TXT is here: https://github.com/dotnet/runtime/blob/main/eng/regenerate-third-party-notices.proj

### CODE REVIEW

File: src/installer/pkg/THIRD-PARTY-NOTICES.TXT
Comment (am11) [inline src/installer/pkg/THIRD-PARTY-NOTICES.TXT:1603]: llvm is referenced above https://github.com/dotnet/runtime/blob/7ed33d80c92fa0b7ae740df60a460e984f2f442b/src/installer/pkg/THIRD-PARTY-NOTICES.TXT#L452

Can we fold it into a single entry?

Comment (NikolaMilosavljevic) [reply src/installer/pkg/THIRD-PARTY-NOTICES.TXT:1603]: We could remove the new entry, if old one covers the `remote stack unwind`.

Comment (am11) [reply src/installer/pkg/THIRD-PARTY-NOTICES.TXT:1603]: We are using several LLVM components across coreclr and mono; combined. I think one license is enough (since the wordings remain the same), and we are not being specific about other usages of LLVM in runtime either.

cc @jkotas

Comment (jkotas) [reply src/installer/pkg/THIRD-PARTY-NOTICES.TXT:1603]: LLVM relicensed their sources.

The old sources are available under legacy LLVM License terms. The existing LLVM entry has the legacy LLVM License terms.

The current sources are available under the current LLVM License terms (Apache with LLVM Exception - https://llvm.org/LICENSE.txt).

We have a mix of both sources licensed under the old terms and the new terms, so I think we should keep both old and new notices. We may want to change the section titles to "License notice for The LLVM Compiler Infrastructure (Legacy License)" and "License notice for The LLVM Project" to make it clear.

File: src/installer/pkg/THIRD-PARTY-NOTICES.TXT
Comment (jkotas) [inline src/installer/pkg/THIRD-PARTY-NOTICES.TXT:1819]: This is third entry for Newtonsoft.Json. Do we need all three?

File: src/installer/pkg/THIRD-PARTY-NOTICES.TXT
Comment (NikolaMilosavljevic) [reply src/installer/pkg/THIRD-PARTY-NOTICES.TXT:1855]: This is coming from `runtime` repo's TPN file, and the root cause might be this: https://github.com/dotnet/runtime/blob/7d1536595aac1a3a2435aa2901d34798d78ca96a/src/libraries/System.Private.CoreLib/src/System/DateTime.cs#L114

Comment (jkotas) [reply src/installer/pkg/THIRD-PARTY-NOTICES.TXT:1855]: My comment is for "Notice for Library of Congress"

File: src/installer/pkg/THIRD-PARTY-NOTICES.TXT
Comment (NikolaMilosavljevic) [reply src/installer/pkg/THIRD-PARTY-NOTICES.TXT]: Fixed with https://github.com/dotnet/runtime/pull/91933/commits/dcda10a320cafc525f5b7f4f82b7d1fbaf123e16

File: src/installer/pkg/THIRD-PARTY-NOTICES.TXT
Comment (NikolaMilosavljevic) [reply src/installer/pkg/THIRD-PARTY-NOTICES.TXT:1824]: Aah, I see - looking...

**Review by jkotas — APPROVED** [review]

**Review by carlossanlop — APPROVED** [review]
Tell-mode. Approving.

@NikolaMilosavljevic are all the suggestions addressed?

@jeffschwMSFT / @artl93 would you like to give it a final look before I merge it?

**Comment by artl93** [comment]
> Tell-mode. Approving.
> 
> @NikolaMilosavljevic are all the suggestions addressed?
> 
> @jeffschwMSFT / @artl93 would you like to give it a final look before I merge it?

@jeffschwMSFT - I don't have context on the license inclusion / changes to comment, so I defer to you if you have more context.

**Review by jeffschwMSFT — APPROVED** [review]
lgtm

**Comment by NikolaMilosavljevic** [comment]
> Tell-mode. Approving.
> 
> @NikolaMilosavljevic are all the suggestions addressed?
> 
> @jeffschwMSFT / @artl93 would you like to give it a final look before I merge it?

All suggestions have been addressed.

**Comment by NikolaMilosavljevic** [comment]
@carlossanlop this can be merged now.

**Pull request merged**
